src/resizing.jl

```diff
@@ -18,118 +18,144 @@ function restrict(A::AbstractArray, region::Union{Dims,Vector{Int}} = coords_spa
+            @nloops $Npost ipost d->indspost[d] begin
+                iout = first(indout)
+                @nloops $Npre ipre d->indspre[d] begin
+                    out[$(Ipre...), iout, $(Ipost...)] = zero(T)
                 end
-                stride_1 = 1
-                @nexprs $N d->(stride_{d+1} = stride_d*size(out,d))
-                @nexprs $N d->offset_d = 0
                 ispeak = true
-                @inbounds @nloops $N i d->(d==1?(1:1):(1:size(A,d))) d->(if d==dim; ispeak=isodd(i_d); offset_{d-1} = offset_d+(div(i_d+1,2)-1)*stride_d; else; offset_{d-1} = offset_d+(i_d-1)*stride_d; end) begin
-                    indx = offset_0
+                for iA in indA
                     if ispeak
-                        for k = 1:size(A,1)
-                            i_1 = k
-                            out[indx+=1] += half*convert(T, @nref $N A i)
+                        @inbounds @nloops $Npre ipre d->indspre[d] begin
+                            out[$(Ipre...), iout, $(Ipost...)] +=
+                                half*convert(T, A[$(Ipre...), iA, $(Ipost...)])
                         end
                     else
-                        for k = 1:size(A,1)
-                            i_1 = k
-                            tmp = quarter*convert(T, @nref $N A i)
-                            out[indx+=1] += tmp
-                            out[indx+strd] = tmp
+                        @inbounds @nloops $Npre ipre d->indspre[d] begin
+                            tmp = quarter*convert(T, A[$(Ipre...), iA, $(Ipost...)])
+                            out[$(Ipre...), iout, $(Ipost...)]   += tmp
+                            out[$(Ipre...), iout+1, $(Ipost...)] = tmp
                         end
                     end
+                    ispeak = !ispeak
+                    iout += ispeak
                 end
             end
         else
             threeeighths = convert(eltype(T), 0.375)
             oneeighth = convert(eltype(T), 0.125)
-            indx = 0
-            if dim == 1
-                z = convert(T, zero(first(A)))
-                @inbounds @nloops $N i d->(d==1 ? (1:1) : (1:size(A,d))) d->(j_d = i_d) begin
-                    c = d = z
-                    for k = 1:size(out,1)-1
-                        a = c
-                        b = d
-                        j_1 = 2*k
-                        i_1 = j_1-1
-                        c = convert(T, @nref $N A i)
-                        d = convert(T, @nref $N A j)
-                        out[indx+=1] = oneeighth*(a+d) + threeeighths*(b+c)
-                    end
-                    out[indx+=1] = oneeighth*c+threeeighths*d
-                end
-            else
-                fill!(out, zero(T))
-                strd = stride(out, dim)
-                stride_1 = 1
-                @nexprs $N d->(stride_{d+1} = stride_d*size(out,d))
-                @nexprs $N d->offset_d = 0
+            z = convert(T, zero(first(A)))
```

Comment:
Not certain, but I think I put it this way originally to support the following:
```julia
julia> a = [rand(3) for i = 1:5]
5-element Array{Array{Float64,1},1}:                                                                                                                                                         
 [0.465388,0.525425,0.753336]                                                                                                                                                                
 [0.890223,0.465896,0.191895]                                                                                                                                                                
 [0.205448,0.896783,0.0923702]                                                                                                                                                               
 [0.538556,0.862554,0.581898]                                                                                                                                                                
 [0.731725,0.887581,0.0116221]

julia> restrict(a, 1)
3-element Array{Array{Float64,1},1}:                                                                                                                                                         
 [0.45525,0.379187,0.424642]                                                                                                                                                                 
 [0.459918,0.780504,0.239633]                                                                                                                                                                
 [0.500502,0.659429,0.151285]
```
Clearly I should add that to the tests.